### ISSUE TITLE
[BUG] Incorrect sort order for exponential notation metrics/params

### ISSUE THREAD

**Issue by gabegrand** [mlflow/mlflow#2956]
I've noticed that metrics and parameters which have been autoconverted to exponential notation via https://github.com/mlflow/mlflow/pull/2022 do not sort correctly in the main experiment table view.

![Screen Shot 2020-06-17 at 3 42 44 PM](https://user-images.githubusercontent.com/10052880/84943102-de177280-b0b1-11ea-8af7-c8aa31b95176.png)

This bug makes it difficult to sort runs by fields that typically have small decimal values like learning rate. Is there an existing solution, or is that a bug that needs to be fixed?

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04.4 LTS
- **MLflow installed from (source or binary)**: source
- **MLflow version (run ``mlflow --version``)**: 1.7.1
- **Python version**: 3.6.6
- **npm version, if running the dev UI**:
- **Exact command to reproduce**:

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

**Comment by dbczumar** [comment]
@gabegrand thank you for pointing this out. This is definitely an issue. Out of curiosity, what happens when you query this data using the MLflow search API with descending or ascending ordering on the metric? Is the ordering correct there? That'll help is narrow down whether or not this is a UI-specific issue.

**Comment by harupy** [comment]
It looks like `learning_rate` is sorted in alphabetical order. Is this because params are sorted as a string?

**Comment by dbczumar** [comment]
@harupy that explanation makes sense, assuming this is a parameter (sounds like `learning_rate` would be a parameter rather than a metric).

**Comment by harupy** [comment]
Metrics are sorted correctly even in exponential notation.

![Screen Shot 2020-06-20 at 14 55 41](https://user-images.githubusercontent.com/17039389/85194296-3d6ec000-b308-11ea-8c3e-2d04509a997f.png)

https://colab.research.google.com/drive/1UlUznVB1bsi_UoMtE4u63FxpYA9E78wF#scrollTo=Vch5HVFSNnfU

**Comment by dbczumar** [comment]
Awesome! I wonder if the lexical ordering for non-exponentional-notation parameters matches the numerical ordering. If it does, we may want to consider sorting parameters by non-exponential values and then convert them to exponential form for display. @harupy can you investigate this?

**Comment by harupy** [comment]
That approach seems to work.

![Screen Shot 2020-06-20 at 15 54 22](https://user-images.githubusercontent.com/17039389/85195529-55e1d900-b30e-11ea-9837-30938dfeba8a.png)

https://colab.research.google.com/drive/1UlUznVB1bsi_UoMtE4u63FxpYA9E78wF?usp=sharing
